src/librbd/io/CopyupRequest.cc

```diff
@@ -439,50 +448,45 @@ void CopyupRequest<I>::copyup() {
 
 template <typename I>
 void CopyupRequest<I>::handle_copyup(int r) {
+  auto cct = m_image_ctx->cct;
   unsigned pending_copyups;
   {
     Mutex::Locker locker(m_lock);
     ceph_assert(m_pending_copyups > 0);
     pending_copyups = --m_pending_copyups;
   }
 
-  ldout(m_ictx->cct, 20) << "oid=" << m_oid << ", "
-                         << "r=" << r << ", "
-                         << "pending=" << pending_copyups << dendl;
+  ldout(cct, 20) << "oid=" << m_oid << ", " << "r=" << r << ", "
+                 << "pending=" << pending_copyups << dendl;
 
-  if (r == -ENOENT) {
-    if (pending_copyups == 0) {
-      // hide the -ENOENT error if this is the last op
-      r = 0;
-    }
-  } else if (r < 0) {
-    lderr(m_ictx->cct) << "failed to copyup object: "
-                       << cpp_strerror(r) << dendl;
+  if (r < 0 && r != -ENOENT) {
+    lderr(cct) << "failed to copyup object: " << cpp_strerror(r) << dendl;
     complete_requests(false, r);
   }
 
   if (pending_copyups == 0) {
-    finish(r);
+    finish(0);
```

Comment:
@dillaman Is it intentional that we finish with `0` here even if `r` is a non-`-ENOENT` error? Just wondering because previously the behavior seemed to be different in this case.

Reply:
We had previously hid the ```-ENOENT``` error from the last copyup result and if you hit an error, we return it to the attached write requests on line 464 (so technically nothing gets called back in that case and only destroys this state machine).